### PR TITLE
Feature/ipehr 169 api get adl14 template

### PR DESCRIPTION
**Pull request by Setest**

### CODE REVIEW

File: src/pkg/api/template.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	tID := c.Param("template_id") // TODO should have its own structure and validation method
+
+	t, err := h.service.GetByID(c, userID, tID)
+	if err != nil {
+		if errors.Is(err, errors.ErrNotFound) {
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
+
+		log.Printf("Template service error: %s", err.Error()) // TODO replace to ErrorF after merge IPEHR-32
```
Comment (SealTV) [inline src/pkg/api/template.go:64]: Why are you write here %s for error? 
It's will be enough write %v and don't call Error() function

File: src/pkg/api/api.go [hunk 2]
```diff
@@ -46,6 +50,11 @@ type API struct {
 func New(cfg *config.Config, infra *infrastructure.Infra) *API {
 	docService := service.NewDefaultDocumentService(cfg, infra)
 	groupAccessService := groupAccess.NewService(docService, cfg.DefaultGroupAccessID, cfg.DefaultUserID)
+
+	opt14 := adl14.NewParser()
+	opt2 := adl2.NewParser()
```
Comment (SealTV) [inline src/pkg/api/api.go]: Seems like it will be better, if put this line into NewService func.

File: src/pkg/docs/model/template.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	Version     string `json:"version"`
+	Concept     string `json:"concept"`
+	ArchetypeID string `json:"archetype_id"`
+	CreatedAt   string `json:"created_timestamp"`
+	VerADL      VerADL
+	Content     []byte
+}
+
+// Version show as which service we should use for parsing template
+type VerADL = string
```
Comment (SealTV) [inline src/pkg/docs/model/template.go]: just `type ADLVer string` ?

File: src/pkg/docs/parser/adl14/parser.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+func NewParser() *Parser {
+	return &Parser{}
+}
+
+func (*Parser) Version() model.VerADL {
+	return model.VerADL1_4
+}
+
+func (*Parser) AllowedType(s string) (model.ADLTypes, error) {
```
Comment (SealTV) [inline src/pkg/docs/parser/adl14/parser.go]: Name of this function sound like it should return bool response.  
```go
func IsAllowedType (t mode.ADLType) bool {
    return t == models.ADLTypeXML || t == models.ADLTypeJOSN
}

File: src/pkg/docs/model/template.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+const (
+	VerADL1_4 VerADL = "adl1.4"
+	VerADL2   VerADL = "adl2"
+)
+
+type ADLTypes = string
```
Comment (SealTV) [inline src/pkg/docs/model/template.go]: Why are you call `enum` type with `s` ending? 
And write just `type ADLType string`

File: src/pkg/docs/service/template/template.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+	for _, p := range parsers {
+		v := p.Version()
+
+		_, ok := ps[v]
+		if ok {
+			log.Fatalf("ADL parser with ver %s already implemented", v)
+		}
+
+		ps[v] = p
```
Comment (SealTV) [inline src/pkg/docs/service/template/template.go]: I think here there are too complicated code.

**Review by SealTV — COMMENTED** [review]
left some comments

File: src/pkg/docs/service/template/template.go [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+func NewService(docService *service.DefaultDocumentService) *Service {
+	ps := make(map[string]ADLParser)
+
+	opt14 := adl14.NewParser()
+	opt2 := adl2.NewParser()
+
+	ps[opt14.Version()] = opt14
+	ps[opt2.Version()] = opt2
```
Comment (SealTV) [inline src/pkg/docs/service/template/template.go]:
```go
ps := map[string]ADLParser{
    opt14.Version(): opt14,
    opt2.Version(): opt2,
}
```

**Review by SealTV — APPROVED** [review]

**Review by 1db1 — APPROVED** [review]

**Pull request merged**